src/common/store/synchronization.slice.ts

```diff
@@ -70,18 +77,17 @@ export const getSynchronizationState = createSelector(
 
 export const getPublicKey = createSelector(getSynchronizationState, (state) => state.publicKey);
 export const getPrivateKey = createSelector(getSynchronizationState, (state) => state.privateKey);
-export const getHandshaken = createSelector(getSynchronizationState, (state) => state.isHandshaken);
+export const getHandshaken = (target: Process) =>
+  createSelector(getSynchronizationState, (state) => state.isHandshaken[target]);
 
-export const getTargetPublicKey = createSelector(
-  getSynchronizationState,
-  (state) => state.targetPublicKey
-);
+export const getTargetPublicKey = (target: Process) =>
+  createSelector(getSynchronizationState, (state) => state.targetPublicKey[target]);
 
 export const postHandshake = createAction(`${sliceName}/postHandshake`);
 
-export function* handshakeSaga(ipc: ReduxIPC) {
+export function* handshakeSaga(processes: Partial<Record<Process, ReduxIPC>>, self: Process) {
   yield all([
-    ipcWorker(ipc),
+    ...Object.keys(processes).map((target) => ipcWorker(processes, target as Process, self)),
```

Comment:
For better type safety, maybe we can use a wrapper function of Object.keys with proper types.
```ts
type ObjectKeys<T> =
  T extends Record<string, unknown> ? (keyof T)[] :
    T extends number ? [] :
      T extends Array<any> | string ? string[] :
        never;

const keys = <T extends Record<string, unknown>>(object: T): ObjectKeys<T> => {
  return Object.keys(object) as ObjectKeys<T>;
}
```